parachain/pallets/dot-app/src/lib.rs

```diff
@@ -0,0 +1,118 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+use frame_system::{self as system, ensure_signed};
+use frame_support::{
+	decl_error, decl_event, decl_module, decl_storage,
+	dispatch::{DispatchError, DispatchResult},
+	transactional,
+	traits::{
+		Get,
+		EnsureOrigin,
+		Currency,
+		ExistenceRequirement::{KeepAlive, AllowDeath},
+	}
+};
+use sp_runtime::traits::StaticLookup;
+use sp_std::prelude::*;
+use sp_core::H160;
+use sp_runtime::{
+	ModuleId,
+	traits::AccountIdConversion,
+	SaturatedConversion,
+};
+
+use artemis_core::{ChannelId, SubmitOutbound};
+
+mod payload;
+use payload::OutboundPayload;
+
+#[cfg(test)]
+mod mock;
+
+#[cfg(test)]
+mod tests;
+
+type BalanceOf<T> = <<T as Config>::Currency as Currency<<T as frame_system::Config>::AccountId>>::Balance;
+
+pub trait Config: system::Config {
+	type Event: From<Event<Self>> + Into<<Self as system::Config>::Event>;
+
+	type Currency: Currency<Self::AccountId>;
+
+	type SubmitOutbound: SubmitOutbound;
+
+	type CallOrigin: EnsureOrigin<Self::Origin, Success=H160>;
+
+	type ModuleId: Get<ModuleId>;
+}
+
+decl_storage! {
+	trait Store for Module<T: Config> as EthModule {
+		/// Address of the peer application on the Ethereum side.
+		Address get(fn address) config(): H160;
+	}
+}
+
+decl_event!(
+    /// Events for the ETH module.
+	pub enum Event<T>
+	where
+		AccountId = <T as system::Config>::AccountId,
+		Balance = BalanceOf<T>
+	{
+		Locked(AccountId, H160, Balance),
+		Unlocked(H160, AccountId, Balance),
+	}
+);
+
+decl_error! {
+	pub enum Error for Module<T: Config> {
+
+	}
+}
+
+decl_module! {
+	pub struct Module<T: Config> for enum Call where origin: T::Origin {
+
+		type Error = Error<T>;
+
+		fn deposit_event() = default;
+
+		#[weight = 0]
+		#[transactional]
+		pub fn lock(origin, channel_id: ChannelId, recipient: H160, amount: BalanceOf<T>) -> DispatchResult {
+			let who = ensure_signed(origin)?;
+
+			T::Currency::transfer(&who, &Self::account_id(), amount, AllowDeath)?;
```

Comment:
do we want to AllowDeath ? I think probably it's fine, but just checking?

Reply:
AllowDeath means that the account will be deleted, including the nonce state. That might allow a replay of transactions that are not yet expired (especially immortal transactions are a problem).

I feel that allowing death here could be problematic, but would need to do a bit more research.

Reply:
Well, even if we don't allow death for our transfer, the account could get itself deleted through some other mechanism right, and so we're still vulnerable to hypothetical attack?

Though I think replay, even if possible, would only fuck over the reborn account rather than the DOTApp/Bridge in any way? My substrate knowledge in this area is limited though so will leave it to u and vincent to resolve